### PR TITLE
feat: Importing of Posts

### PR DESCRIPTION
**Pull request by Mar0xy**
- Supports Instagram, Mastodon/Pleroma/Akkoma, Twitter and *key

### CODE REVIEW

File: packages/backend/src/queue/processors/ImportNotesProcessorService.ts [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+		private mfmService: MfmService,
+		private apNoteService: ApNoteService,
+		private driveService: DriveService,
+		private downloadService: DownloadService,
+		private queueLoggerService: QueueLoggerService,
+	) {
+		this.logger = this.queueLoggerService.logger.createSubLogger('import-notes');
+	}
+	@bindThis
+	private async _keepTweet(tweet: any) {
```
Comment (dakkar) [inline packages/backend/src/queue/processors/ImportNotesProcessorService.ts:46]: this feels like a hack added to simplify testing during development… consider a comment explaining what this function is for, or maybe remove it?

File: packages/backend/src/queue/processors/ImportNotesProcessorService.ts [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+	private async uploadFiles(dir: any, user: any) {
+		const fileList = fs.readdirSync(dir);
+		for (const file of fileList) {
+			const name = `${dir}/${file}`;
+			if (fs.statSync(name).isDirectory()) {
+				await this.uploadFiles(name, user);
+			} else {
+				const exists = await this.driveFilesRepository.findOneBy({ name: file, userId: user.id });
+
+				if (file.endsWith('.srt')) return;
```
Comment (dakkar) [inline packages/backend/src/queue/processors/ImportNotesProcessorService.ts:64]: why do we dislike subtitles? 😁

File: packages/backend/src/queue/processors/ImportNotesProcessorService.ts [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+	private isIterable(obj: any) {
+		if (obj == null) {
+			return false;
+		}
+		return typeof obj[Symbol.iterator] === 'function';
+	}
+
+	@bindThis
+	public async process(job: Bull.Job<DbNoteImportJobData>): Promise<void> {
+		this.logger.info(`Importing following of ${job.data.user.id} ...`);
```
Comment (dakkar) [inline packages/backend/src/queue/processors/ImportNotesProcessorService.ts:87]: "following"?

Comment (Mar0xy) [reply packages/backend/src/queue/processors/ImportNotesProcessorService.ts:87]: This was from when I copied the following import process as a template and built off it I will remove it later

File: packages/backend/src/queue/processors/ImportNotesProcessorService.ts (hunk 2 above)
Comment (Mar0xy) [reply packages/backend/src/queue/processors/ImportNotesProcessorService.ts:64]: Cause they are part of some videos from IG but you can't really do anything with them on MK.

File: packages/backend/src/queue/processors/ImportNotesProcessorService.ts (hunk 1 above)
Comment (Mar0xy) [reply packages/backend/src/queue/processors/ImportNotesProcessorService.ts:46]: This was already talked about being removed

File: packages/backend/src/queue/processors/ImportNotesProcessorService.ts [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+		}
+
+		if (toot.object.attachment && this.isIterable(toot.object.attachment)) {
+			for await (const file of toot.object.attachment) {
+				const slashdex = file.url.lastIndexOf('/');
+				const name = file.url.substring(slashdex + 1);
+				const exists = await this.driveFilesRepository.findOneBy({ name: name, userId: user.id });
+				if (exists) {
+					files.push(exists);
+				}
```
Comment (dakkar) [inline packages/backend/src/queue/processors/ImportNotesProcessorService.ts:292]: why don't we try downloading attachments for mastodon imports?

Comment (Mar0xy) [reply packages/backend/src/queue/processors/ImportNotesProcessorService.ts:292]: They are already locally provided in the zip see https://github.com/transfem-org/Sharkey/blob/b60295e46dcf4314e1f14a449b2904d9622495c4/packages/backend/src/queue/processors/ImportNotesProcessorService.ts#L180

**Pull request merged**